chains/evm/voter/proposal.go

```diff
@@ -87,27 +87,34 @@ func (p *Proposal) Execute(client ChainClient) error {
 	if err != nil {
 		return err
 	}
-	gasLimit := uint64(2000000)
-	gp, err := client.GasPrice()
+	// gasLimit := uint64(2000000)
+	// gp, err := client.GasPrice()
+	// if err != nil {
+	// 	return err
+	// }
+	// client.LockNonce()
+	// n, err := client.UnsafeNonce()
+	// if err != nil {
+	// 	return err
+	// }
+	client.LockOpts()
+	opts, err := client.UnsafeOpts()
 	if err != nil {
 		return err
 	}
-	client.LockNonce()
```

Comment:
Do not you think that we should remove nonceMutex since it is not used any more and could be confusing others. Also im thinking about making UnsafeNonce adn UnsafeIncreaseNonce private methods that could be used only inside UnsafeOpts, add UnsafeUpdate nonce for increasing nonce...

Reply:
Yup definitely just hadn't removed it yet. It is removed now. I made UnsafeNonce private, but kept UnsafeIncreaseNonce a public method for now as we only have confirmation of the transaction being successful within `proposal.go` and the nonce is the only opts field needing to be updated after a transaction send.